src/main/java/seedu/address/model/person/Phone.java

```diff
@@ -62,6 +63,17 @@ public boolean exactEquals(Phone otherPhone) {
 
     @Override
     public int hashCode() {
-        return phone.hashCode();
+        return removeAllWhiteSpace(phone).hashCode();
     }
+
+    /**
+     * Removes all white space from the given string.
+     *
+     * @param str String to remove all white space from.
+     * @return String with all white space removed.
+     */
+    private String removeAllWhiteSpace(String str) {
+        return str.replaceAll(" ", "");
```

Comment:
To be safe, consider changing this to consider all types of whitespaces `\s`? (Since the validation regex allows all types of white spaces)

Reply:
Upon further thought, perhaps it's better to only allow ` ` instead of the more general `\s` as the validation regex.